cdap-spark-core/src/main/java/co/cask/cdap/app/runtime/spark/SparkRuntimeService.java

```diff
@@ -225,9 +227,16 @@ public SparkExecutionContext create(SparkRuntimeContext runtimeContext) {
 
         // Localize the hConf file to executor nodes
         localizeResources.add(new LocalizeResource(saveHConf(hConf, tempDir)));
+
+        List<File> extraJarFiles = CConfigurationUtil.getExtraJars(cConf);
+        for (File jar : extraJarFiles) {
+          String jarName = jar.getName();
+          extraJars.add(jarName);
+          localizeResources.add(new LocalizeResource(new File(jarName)));
```

Comment:
because original files were created with paths, but here we just need the filename.